File: src/Scripting/PHP/bridge.php
<?php

interface Device
{
    public function powerOn(): string;
    public function mute(): string;
}

final class TvDevice implements Device
{
    public function powerOn(): string { return 'TV:on'; }
    public function mute(): string { return 'TV:muted'; }
}

final class RadioDevice implements Device
{
    public function powerOn(): string { return 'Radio:on'; }
    public function mute(): string { return 'Radio:muted'; }
}

abstract class RemoteControl
{
    public function __construct(protected Device $device) {}
    abstract public function activate(): string;
}

final class BasicRemote extends RemoteControl
{
    public function activate(): string { return $this->device->powerOn(); }
}

final class MuteRemote extends RemoteControl
{
    public function activate(): string { return $this->device->mute(); }
}

$tv = new TvDevice();
$radio = new RadioDevice();
echo 'basic-tv=' . (new BasicRemote($tv))->activate() . PHP_EOL;
echo 'basic-radio=' . (new BasicRemote($radio))->activate() . PHP_EOL;
echo 'mute-tv=' . (new MuteRemote($tv))->activate() . PHP_EOL;
echo 'mute-radio=' . (new MuteRemote($radio))->activate() . PHP_EOL;

File: src/Web/Dart/bridge.dart
abstract interface class Device {
  String powerOn();
  String mute();
}

class TvDevice implements Device {
  @override
  String powerOn() => 'TV:on';
  @override
  String mute() => 'TV:muted';
}

class RadioDevice implements Device {
  @override
  String powerOn() => 'Radio:on';
  @override
  String mute() => 'Radio:muted';
}

abstract class RemoteControl {
  RemoteControl(this.device);
  final Device device;
  String activate();
}

class BasicRemote extends RemoteControl {
  BasicRemote(super.device);
  @override
  String activate() => device.powerOn();
}

class MuteRemote extends RemoteControl {
  MuteRemote(super.device);
  @override
  String activate() => device.mute();
}

void main() {
  final tv = TvDevice();
  final radio = RadioDevice();
  print('basic-tv=${BasicRemote(tv).activate()}');
  print('basic-radio=${BasicRemote(radio).activate()}');
  print('mute-tv=${MuteRemote(tv).activate()}');
  print('mute-radio=${MuteRemote(radio).activate()}');
}
